libcloud/storage/drivers/azure_blobs.py

```diff
@@ -929,6 +932,31 @@ def delete_object(self, obj):
 
         return False
 
+    def _fix_headers(self, headers):
```

Comment:
Can you please add some unit tests for this method / functionality?

Reply:
Fitting this into the mock request/response framework was proving challenging so I added an isolated test for the new normalization functionality in 98cd383. I also added an integration test for this scenario in https://github.com/c-w/libcloud-tests/commit/e305b9d34276f50543b8c3a5f342eb18f04fe945.

Reply:
Investigating why the new e2e test fails for Azurite v3 only ([see test run](https://clewolff.visualstudio.com/libcloud-tests/_build/results?buildId=1593&view=logs&j=2be97940-aaf3-5b4e-dc43-b88617d20841&t=741988fb-4d4e-5131-94f3-05b5b69d36ed)).

Reply:
Yeah, direct unit tests for that method are fine :)

Maybe something to do with auth signature? Since, IIRC, we use header values when generating the signature.

Reply:
On that note - I wonder if we could some merge some of that end to end tests in that repo (if it makes sense and how we could do that).

Reply:
I believe the test failure may be linked to an issue in AzuriteV3, see https://github.com/Azure/Azurite/issues/629#issuecomment-776258485. Given that the test passes for all other storage targets, I disabled the test for AzuriteV3 in https://github.com/c-w/libcloud-tests/commit/b2be083358cfe3898aff50bb380c175cbf8ea2aa.